src/algebraic_geometry/elliptic_curve/weierstrass.lean

```diff
@@ -365,13 +365,17 @@ begin
   any_goals { rw [degree_add_eq_right_of_degree_lt]; simp only [h]; dec_trivial }
 end
 
-/-- The coordinate ring $R[W] := R[X, Y] / \langle W(X, Y) \rangle$ of `W`. -/
-@[reducible] def coordinate_ring : Type u := adjoin_root W.polynomial
+/-- The coordinate ring $R[W] := R[X, Y] / \langle W(X, Y) \rangle$ of `W`.
+
+Note that `derive` generates a reducible instance of `comm_ring` for `coordinate_ring`, which in
+certain circumstances is extremely slow as all its types are rechecked for equality, in which case
+add `attribute [irreducible] coordinate_ring.comm_ring` locally to block type-level unification. -/
```

Comment:
Let's be a bit more precise where the issues come from, and some pointers on how to resolve this hopefully.
```suggestion
Note that `derive` generates a reducible instance of `comm_ring` for `coordinate_ring`, which in
certain circumstances is extremely slow when all instances in its definition are unified
exponentially many times, in which case add `attribute [irreducible] coordinate_ring.comm_ring`
locally to block type-level unification.

TODO Lean 4: verify if the new def-eq cache (lean4#1102) fixed this issue.

See Zulip thread: https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Why.20is.20class_group.2Emk.20so.20slow.3F
-/
```